_posts/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm.md

```diff
@@ -0,0 +1,258 @@
+
+
+
+### Activation Function
+
+I'm using a sigmoid function for my activation function ([here](https://github.com/cohen990/evolution-of-tetris/blob/3147c87c84aafce0d3ed21d9b21b0b0ebae92a30/src/java/com/cohen990/Network.java#L47)). A sigmoid function is bounded between 0 and 1 and is non-linear. This created a "decision boundary" for the node. More information is available [here](https://stackoverflow.com/questions/9782071/why-must-a-nonlinear-activation-function-be-used-in-a-backpropagation-neural-net).
+
+### Weights and Biases
+
+[![diagram of weights and biases in a neural network]({{site.baseurl}}/assets/img/custom/blog/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm/weights-biases.jpg)]({{site.baseurl}}/assets/img/custom/blog/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm/weights-biases.jpg)
+
+Each connection in a neural network has a weight. It is factored in to the calculation of the value of the node it is linking to. Each node, not including the input layer, has a bias. 
+
+The network is defined by the structure of the layers, the connections between them and their weights and biases. The structure and connections remain constant throughout an experiment but the weights and the biases will change.
+
+When you talk about "training a neural network" you are talking about modifying the weights and biases to produce better results. They are the key.
+
+### Evolutionary Algorithm
+
+In my experiment, I used an evolutionary algorithm to train the network. the evolutionary algorithm (EA) is not the most efficient, but it is easy to conceptualise. That's why I chose it.
+
+The fundamental principle of EA is generations. Each generation is a group of networks that are all different. You allow each network to play through a game of tetris and then you eliminate the worst of them. The surviving population is allowed to reproduce. The idea is that the surviving networks are more likely to perform better.
+
+Of course, it wouldn't be evolutionary without evolution. So there must be mutations. This is how EA allows networks to progress and perform better and better.
+
+My implementation of EA is unusual in that each network has only one parent. The parent network is duplicated and passed through a series of mutations. The child network has a chance of performing better or worse than the parent. Because the worst performers are eliminated, the average performance of the population increases over time as the positive changes to the networks are compounded.  
+
+It's very common for EA to make use of a crossover algorithm. which allows a child to have multiple parents. It is a way of allowing features developed in one network to be combined with features developed in another network. This is the real magic of EA. Unfortunately I was not aware of it at the start of my experiment.
+
+### Fitness Function
```

Comment:
This renders in an unexpected way. The following headings look like a start of a paragraph, while it reads more like a list of things. It may help to make it into a list perhaps ? Also - having a screenshot/picture with these labeled may help